### PR TITLE
PLT-6595 (Client): Elasticsearch indexing system console UI

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
System console UI for Elasticsearch indexing job.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6595

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file updates

### CODE REVIEW

**Comment by grundleborg** [comment]
@jasonblais the elasticsearch testing spinmint has been updated with AM's changes, so is ready for testing again now.

**Comment by lfbrock** [comment]
@asaadmahmood - I think the spacing beside the Cancel button is still small? 

screenshot 2017-07-24 10 49 03

File: webapp/components/admin_console/elasticsearch_status/status.jsx [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+        );
+    };
+
+    getChosenJob = () => {
+        let chosenJob = null;
+
+        if (this.props.jobs.length > 0) {
+            this.props.jobs.forEach((job) => {
+                if (job.status === JobStatuses.CANCEL_REQUESTED || job.status === JobStatuses.IN_PROGRESSZ) {
+                    chosenJob = job;
```
Comment (saturninoabril) [inline webapp/components/admin_console/elasticsearch_status/status.jsx]: Just a typo error: `IN_PROGRESSZ`

File: webapp/components/admin_console/elasticsearch_status/status.jsx [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+                );
+                statusColor = '#eac262';
+            } else if (chosenJob.status === JobStatuses.CANCELED) {
+                status = (
+                    <FormattedMessage
+                        id='admin.elasticsearchStatus.statusCancelled'
+                        defaultMessage='Indexing job cancelled.'
+                    />
+                );
+                statusColor = '##ea6262';
```
Comment (saturninoabril) [inline webapp/components/admin_console/elasticsearch_status/status.jsx]: Typo error here: `##ea6262`

Would it be possible to move this hardcoded `statusColor` somewhere to *.scss? e.g.: 
```
.job-cancel {
    color: #ea6262;
} 
```

File: webapp/components/admin_console/elasticsearch_status/status.jsx [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+                statusHelp = (
+                    <FormattedMessage
+                        id='admin.elasticsearchStatus.statusError.help'
+                        defaultMessage='Mattermost encountered an error building the Elasticsearch index: {error}'
+                        values={{
+                            error: chosenJob.data ? (chosenJob.data.error || '') : ''
+                        }}
+                    />
+                );
+                statusColor = '##ea6262';
```
Comment (saturninoabril) [inline webapp/components/admin_console/elasticsearch_status/status.jsx]: Typo error here: `##ea6262`

File: webapp/components/admin_console/elasticsearch_status/status.jsx [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+    };
+
+    getChosenJob = () => {
+        let chosenJob = null;
+
+        if (this.props.jobs.length > 0) {
+            this.props.jobs.forEach((job) => {
+                if (job.status === JobStatuses.CANCEL_REQUESTED || job.status === JobStatuses.IN_PROGRESS) {
+                    chosenJob = job;
+                    return false;
```
Comment (hmhealey) [inline webapp/components/admin_console/elasticsearch_status/status.jsx]: Returning true or false does nothing here. The Mozilla docs recommend using an array.prototype.find here

File: webapp/components/admin_console/elasticsearch_status/status.jsx [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+                }
+                return true;
+            });
+
+            if (!chosenJob) {
+                this.props.jobs.forEach((job) => {
+                    if (job.status !== JobStatuses.PENDING && chosenJob) {
+                        return false;
```
Comment (hmhealey) [inline webapp/components/admin_console/elasticsearch_status/status.jsx]: Same here

File: webapp/components/admin_console/elasticsearch_status/status.jsx [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+            () => {
+                this.reload();
+            },
+            () => {
+                this.reload();
+            }
+        );
+    };
+
+    getChosenJob = () => {
```
Comment (hmhealey) [inline webapp/components/admin_console/elasticsearch_status/status.jsx]: What does this mean by the "chosen" job? This doesn't look like it's selected by the user

Comment (grundleborg) [inline webapp/components/admin_console/elasticsearch_status/status.jsx]: It's the job that has been "chosen" to use to display the status of the indexer. It's a hack until we have a proper job UI as a table.

**Comment by jasonblais** [comment]
Closing and re-opening to re-trigger the build.

**Comment by jasonblais** [comment]
@grundleborg Looks like the build is failing. Not sure if it's just a random unit test failure or an actual issue.

**Pull request merged**